resources/assets/lib/profile-page/top-scores.tsx

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import GameMode from 'interfaces/game-mode';
+import ScoreJson from 'interfaces/score-json';
+import { route } from 'laroute';
+import PlayDetailList from 'play-detail-list';
+import ExtraHeader from 'profile-page/extra-header';
+import * as React from 'react';
+import ShowMoreLink from 'show-more-link';
+import ExtraPageProps, { ProfilePagePaginationData, TopScoreSection } from './extra-page-props';
+
+interface SectionMap {
+  count: 'scores_best_count' | 'scores_first_count';
+  key: TopScoreSection;
+  type: string;
+}
+
+const sectionMaps: SectionMap[] = [
+  {
+    count: 'scores_best_count',
+    key: 'scoresBest',
+    type: 'best',
+  },
+  {
+    count: 'scores_first_count',
+    key: 'scoresFirsts',
+    type: 'firsts',
+  },
+];
+
+type ScoreData = ScoreJson[] | { error: string };
+
+type Props = {
+  currentMode: GameMode;
+  pagination: ProfilePagePaginationData;
+  scoresNotice?: string;
+} & {
+  [key in TopScoreSection]?: ScoreData;
+} & ExtraPageProps;
+
+export default class TopScores extends React.PureComponent<Props> {
+  render() {
+    return (
+      <div className='page-extra'>
+        <ExtraHeader name={this.props.name} withEdit={this.props.withEdit} />
+
+        {this.props.scoresNotice != null && (
+          <div className='wiki-notice'>
+            <span className='fas fa-exclamation-circle' />
+            {' '}
+            <div
+              className='wiki-notice__markdown-inline-content'
+              dangerouslySetInnerHTML={{ __html: this.props.scoresNotice }}
+            />
+          </div>
+        )}
+
+        {sectionMaps.map((section) => (
+          <div key={section.key}>
+            <h3 className='title title--page-extra-small'>
+              {osu.trans('users.show.extra.top_ranks.best.title')}
```

Comment:
Needs some variable.
Also the param is `firsts` but the translation key is `first` (and other general mismatches) 🤔

Reply:
fun ಠ_ಠ

I'm not fixing that now